src/Sulu/Bundle/AdminBundle/Resources/js/components/BlockCollection/tests/BlockCollection.test.js

```diff
@@ -406,3 +406,170 @@ test('Should apply renderBlockContent before rendering the block content includi
+    };
+
+    const value = [
+        {
+            type: 'type1',
+            content: 'Test 1',
+        },
+        {
+            type: 'type2',
+            content: 'Test 2',
+        },
+        {
+            type: 'type1',
+            content: 'Test 3',
+        },
+    ];
+
+    const blockCollection = mount(
+        <BlockCollection
+            defaultType="editor"
+            onChange={jest.fn()}
+            renderBlockContent={jest.fn()}
+            types={types}
+            value={value}
+        />
+    );
+
+    blockCollection.instance().expandedBlocks[0] = true;
+
+    expect(blockCollection.props().value.length).toBe(3);
+    expect(blockCollection.instance().expandedBlocks.length).toBe(3);
+    expect(blockCollection.instance().generatedBlockIds.length).toBe(3);
+    expect(blockCollection.instance().expandedBlocks[0]).toBe(true);
+
+    blockCollection.setProps({
+        value: [
+            {
+                type: 'type1',
+                content: 'Test 1',
+            },
+        ],
+    });
+
+    expect(blockCollection.props().value.length).toBe(1);
+    expect(blockCollection.instance().expandedBlocks.length).toBe(1);
+    expect(blockCollection.instance().generatedBlockIds.length).toBe(1);
+    expect(blockCollection.instance().expandedBlocks[0]).toBe(true);
+});
+
+test('Should adjust expandedBlocks and generatedBlockIds after updating the value variable with more entries', () => {
+    const types = {
+        type1: 'Type 1',
+        type2: 'Type 2',
+    };
+
+    const value = [
+        {
+            type: 'type1',
+            content: 'Test 1',
+        },
+    ];
+
+    const blockCollection = mount(
+        <BlockCollection
+            defaultType="editor"
+            onChange={jest.fn()}
+            renderBlockContent={jest.fn()}
+            types={types}
+            value={value}
+        />
+    );
+
+    blockCollection.instance().expandedBlocks[0] = true;
+
+    expect(blockCollection.props().value.length).toBe(1);
+    expect(blockCollection.instance().expandedBlocks.length).toBe(1);
+    expect(blockCollection.instance().generatedBlockIds.length).toBe(1);
+    expect(blockCollection.instance().expandedBlocks[0]).toBe(true);
+
+    blockCollection.setProps({
+        value: [
+            {
+                type: 'type1',
+                content: 'Test 1',
+            },
+            {
+                type: 'type2',
+                content: 'Test 2',
+            },
+            {
+                type: 'type1',
+                content: 'Test 3',
+            },
+        ],
+    });
+
+    expect(blockCollection.props().value.length).toBe(3);
+    expect(blockCollection.instance().expandedBlocks.length).toBe(3);
+    expect(blockCollection.instance().generatedBlockIds.length).toBe(3);
+    expect(blockCollection.instance().expandedBlocks[0]).toBe(true);
+});
+
+test('Should not adjust expandedBlocks and generatedBlockIds after updating the value variable with the same ' +
```

Comment:
This function misses the closing parantheses. And can we slight rename it, so that everything fits in one line? Using this `+` looks weird :see_no_evil: How about something more concise like:

Updating value with same length should not adjust expandedBlocks and generatedBlockIds.

Reply:
Fixed, thanks for the better naming :)